editor/components/CrashSwapAddressButton.tsx

```diff
@@ -1,58 +1,55 @@
 
 /**
- * Button on the primary address data card that allows users to swap
- * the primary and secondary addresses of a crash
+ * Button on the crash address edit modal that swaps the inputs of the
+ * primary and secondary address fields in the form
  */
-export default function CrashSwapAddressButton<
-  T extends Record<string, unknown>,
->({ record, mutation, onSaveCallback }: HeaderActionComponentProps<T>) {
-  // switching all primary and secondary address values here
-  const mutationVariables = {
-    rpt_block_num: record.rpt_sec_block_num,
-    rpt_street_name: record.rpt_sec_street_name,
-    rpt_street_desc: record.rpt_sec_street_desc,
-    rpt_road_part_id: record.rpt_sec_road_part_id,
-    rpt_rdwy_sys_id: record.rpt_sec_rdwy_sys_id,
-    rpt_hwy_num: record.rpt_sec_hwy_num,
-    rpt_street_pfx: record.rpt_sec_street_pfx,
-    rpt_street_sfx: record.rpt_sec_street_sfx,
-    rpt_sec_block_num: record.rpt_block_num,
-    rpt_sec_street_name: record.rpt_street_name,
-    rpt_sec_street_desc: record.rpt_street_desc,
-    rpt_sec_road_part_id: record.rpt_road_part_id,
-    rpt_sec_rdwy_sys_id: record.rpt_rdwy_sys_id,
-    rpt_sec_hwy_num: record.rpt_hwy_num,
-    rpt_sec_street_pfx: record.rpt_street_pfx,
-    rpt_sec_street_sfx: record.rpt_street_sfx,
-  };
+export default function CrashSwapAddressButton({
+  getValues,
+  setValue,
+}: {
+  getValues: UseFormGetValues<AddressFormInputs>;
+  setValue: UseFormSetValue<AddressFormInputs>;
+}) {
+  const handleSwapAddresses = () => {
+    // Get the current values in the form state
+    const currentValues = getValues();
 
-  const { mutate, loading: isMutating } = useMutation(mutation);
+    // Swap the primary and secondary values
+    const swappedValues = {
+      rpt_block_num: currentValues.rpt_sec_block_num,
+      rpt_street_pfx: currentValues.rpt_sec_street_pfx,
+      rpt_street_name: currentValues.rpt_sec_street_name,
+      rpt_street_sfx: currentValues.rpt_sec_street_sfx,
+      rpt_street_desc: currentValues.rpt_sec_street_desc,
+      rpt_road_part_id: currentValues.rpt_sec_road_part_id,
+      rpt_rdwy_sys_id: currentValues.rpt_sec_rdwy_sys_id,
+      rpt_hwy_num: currentValues.rpt_sec_hwy_num,
+
+      rpt_sec_block_num: currentValues.rpt_block_num,
+      rpt_sec_street_pfx: currentValues.rpt_street_pfx,
+      rpt_sec_street_name: currentValues.rpt_street_name,
+      rpt_sec_street_sfx: currentValues.rpt_street_sfx,
+      rpt_sec_street_desc: currentValues.rpt_street_desc,
+      rpt_sec_road_part_id: currentValues.rpt_road_part_id,
+      rpt_sec_rdwy_sys_id: currentValues.rpt_rdwy_sys_id,
+      rpt_sec_hwy_num: currentValues.rpt_hwy_num,
+    };
+
+    // Set the new swapped values in the form state
+    Object.entries(swappedValues).forEach(([key, value]) => {
+      // Setting shouldDirty on setValue allows us to enable the save button when addresses are swapped
+      setValue(key as keyof AddressFormInputs, value, { shouldDirty: true });
```

Comment:
this was a cool [TIL](https://github.com/react-hook-form/react-hook-form/issues/72#issuecomment-656500986), previously if you made and edit and saved and then opened modal and reverted the edit, the form thought it wasnt dirty so the save button was disabled